src/store/kv-block-store.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * AR.IO Gateway
+ * Copyright (C) 2022-2023 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import winston from 'winston';
+
+import { jsonBlockToMsgpack, msgpackToJsonBlock } from '../lib/encoding.js';
+import { KVBufferStore, PartialJsonBlockStore } from '../types.js';
+import { PartialJsonBlock } from '../types.js';
+
+export class KvTransactionStore implements PartialJsonBlockStore {
+  private log: winston.Logger;
+  private kvBufferStore: KVBufferStore;
+
+  constructor({
+    log,
+    kvBufferStore,
+  }: {
+    log: winston.Logger;
+    kvBufferStore: KVBufferStore;
+  }) {
+    this.log = log.child({ class: this.constructor.name });
+    this.kvBufferStore = kvBufferStore;
+  }
+
+  private blockHash(hash: string) {
+    return `${hash}-hash`;
+  }
+
+  private blockHeight(height: number) {
+    return `${height}-height`;
+  }
```

Comment:
@djwhitt - unlikely for collisions using `hash` and `height` as the keys for the block store, so if you feel these are unnecessary can remove

Reply:
@dtfiedler - eh, I kinda like it for clarity + key sorting, but let's make them prefixes (so they sort together) and shorten them to `#-` (for hash) and `H-` for height (to reduce key size).

Reply:
@dtfiedler Let's go with `|` instead of `-` too so that the delimiter isn't in the set of base64url characters.